.github/workflows/build_packages.yml

```diff
@@ -39,6 +43,11 @@ jobs:
           ./build_tools/compute_local_version.py -rc --write-json
           ./build_tools/build_release.py --no-download
 
+      - name: Build C++ dependencies (Linux x86_64)
```

Comment:
This misses
```
      env:
        OVERRIDE_PYTHON_VERSIONS: "${{ matrix.python-version }}"
```
see for example https://github.com/nod-ai/shark-ai/blob/main/.github/workflows/build_packages.yml.

That is also the reason why you want to spin the matrix. If you don't override here, there is no need for the matrix and that is what confused me during the first round of reviews.